preprocess/bom_test.go

```diff
@@ -58,3 +58,17 @@ func TestRemoveBOM(t *testing.T) {
 		require.True(t, bytes.HasPrefix(n, []byte("#!")), "%s does not start with shebang: %#v", fn, n)
 	}
 }
+
+func Test_encodeToUTF8(t *testing.T) {
```

Comment:
This feels like a rat hole. 😦 So many ways to be wrong, and we are very forgiving.